typescript/packages/subsurface-viewer/src/components/Map.tsx

```diff
@@ -504,8 +424,20 @@ const Map: React.FC<MapProps> = ({
         undefined
     );
 
+    // Get vertical scaling factor defined in viewports.
+    const verticalScale = useVerticalScale(views?.viewports);
+
     // Used for scaling in z direction using arrow keys.
-    const [zScale, updateZScale] = React.useReducer(updateZScaleReducer, 1);
+    const { zScale: zReScale, divRef: zScaleRef } = useHandleRescale();
+
+    const { shiftHeld, divRef: shiftHeldRef } = useShiftHeld();
+
+    const divRef = mergeRefs(
+        zScaleRef,
+        shiftHeldRef
+    ) as React.Ref<HTMLDivElement>;
+
+    const zScale = verticalScale ?? zReScale;
```

Comment:
Does that mean that if any viewport has a `verticalScale` set, the arrow keys are disabled ?
In that case, we probably want to disable the event handling, no ?

Does this allow the client application to set its own handler of key events ?

Reply:
> Does that mean that if any viewport has a `verticalScale` set, the arrow keys are disabled ? In that case, we probably want to disable the event handling, no ?

Yes, the arrow keys will be disabled. Will add a change to disable the events.

> 
> Does this allow the client application to set its own handler of key events ?

That was the plan initially, to add let `verticalScale` be something like `number | () => number` and, if not set, fall back to a default value like the current event handler hook. So probably in an upcoming PR.